homeassistant/components/ntfy/notify.py

```diff
@@ -0,0 +1,153 @@
+"""ntfy platform for notify component."""
+from http import HTTPStatus
+import logging
+from typing import Any
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    ATTR_MESSAGE,
+    ATTR_TITLE,
+    ATTR_TITLE_DEFAULT,
+    PLATFORM_SCHEMA,
+    BaseNotificationService,
+)
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_URL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+CONF_TOPIC = "topic"
+DEFAULT_URL = "https://ntfy.sh"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(
+            CONF_TOPIC
+        ): cv.string,  # have to set topic globally in the configuration, can override in "data"
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_URL, default=DEFAULT_URL): cv.url,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(
+    hass: HomeAssistant,
+    config: ConfigType,
+    discovery_info=None,  # pylint: disable=hass-argument-type
+) -> BaseNotificationService:
+    """Get the ntfy notification service."""
+    url = config.get(CONF_URL)
+    topic = config.get(CONF_TOPIC)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    verify_ssl = config.get(CONF_VERIFY_SSL)
+
+    if username and password:
+        auth = requests.auth.HTTPBasicAuth(username, password)
+    else:
+        auth = None
+
+    return NtfyNotificationService(
+        hass,
+        url,
+        topic,
+        auth,
+        verify_ssl,
+    )
```

Comment:
I'd pass these by kwarg names (and maybe make the constructor kwarg-only past `hass`, i.e. `(self, hass, *, url, ...)`